uniffi_bindgen/src/interface/mod.rs

```diff
@@ -216,75 +216,63 @@ impl<'ci> ComponentInterface {
         self.types.get_type_definition(name)
     }
 
-    /// Check whether the given type contains any (possibly nested) Type::Object references.
+    /// Iterate over all the types found within in the given item.
+    /// Unlike the `IterTypes::iter_types` method, this implementation is able to
+    /// recurse into the definitions of user-defined types like records and enums.
+    fn iter_types_in<'a, T: IterTypes>(&'a self, item: &'a T) -> TypeIterator<'a> {
+        Box::new(
+            item.iter_types()
+                .map(move |t| {
+                    let nested_types = match t {
+                        Type::Object(nm) => {
+                            self.get_object_definition(nm).map(IterTypes::iter_types)
+                        }
+                        Type::Error(nm) => self.get_error_definition(nm).map(IterTypes::iter_types),
+                        Type::Enum(nm) => self.get_enum_definition(nm).map(IterTypes::iter_types),
+                        Type::Record(nm) => {
```

Comment:
Hrm, actually this probably needs to recurse through `iter_types_in` rather than just `iter_types`, but then it might trigger pathological cases of infinite recursion...

Reply:
Indeed, we have this unbounded recursion in the version of this currently on `main`, ref https://github.com/mozilla/uniffi-rs/issues/1006

Reply:
I've pushed an update the resolves the infinite-recursion problem.